command-acceptor/app/controllers/implementations/Mediator.py

```diff
@@ -3,29 +3,48 @@
 from app.controllers.implementations.CommandFactory import CommandFactory
 from model.AcceptedMessage import AcceptedMessage
 from model.Command import Command
+from model.ErrorMessage import ErrorMessage
 from threading import Event
+from app.controllers.implementations.ProcessorQueuePusher import ProcessorQueuePusher
+from app.controllers.implementations.ErrorQueuePusher import ErrorQueuePusher
+from model.SentMessageContainer import SentMessageContainer
+from model.ErrorMessageContainer import ErrorMessageContainer
 
 
 class Mediator(interfaces.Mediator):
-    def __init__(self, event: Event, message_holder: MessageHolder)-> None:
+    def __init__(self, event: Event, message_holder: MessageHolder, processor_queue_pusher: ProcessorQueuePusher,
+                 error_queue_pusher: ErrorQueuePusher) -> None:
         self.busy: bool = False
         self.event = event
         self.message_holder = message_holder
+        self.processor_queue_pusher = processor_queue_pusher
+        self.error_queue_pusher = error_queue_pusher
+        print(7)
 
     # def is_busy(self) -> bool:
     #     return self.busy
 
-    def start(self)-> None:
+    def start(self) -> None:
         self.busy = True
+        print(5)
         while self.busy:
```

Comment:
With this logic if we have a lot of messages to process we will be not able to stop the component till we sent everything. Which is not ideal. It should be vice versa driven, we have message? yes, send it, do we still work, yes, send next one, no? stop processing messages

Reply:
I have fixed this by adding one more condition to the second while loop